crypto/src/test_utils/test_conversation.rs

```diff
@@ -47,6 +49,72 @@ impl<'a> TestConversation<'a> {
             .await
             .expect("all invitations succeeded");
     }
+
+    /// All members of this conversation.
+    ///
+    /// The creator is always the first member returned.
+    pub fn members(&self) -> impl Iterator<Item = &SessionContext> {
+        std::iter::once(self.creator).chain(self.joiners.iter().copied())
+    }
+}
+
+/// This struct encapsulates the result of a join by external commit.
+///
+/// To notify all existing members of the conversation, call [`Self::notify_existing_members`].
+/// Otherwise, use the struct members to do things manually.
+pub struct ExternalJoinGuard<'a> {
```

Comment:
Shouldn't this also be `pub(crate)`?

Reply:
We could I suppose, but given that this whole module gets compiled only in `cfg(test)`, it feels kind of silly to split hairs. `cfg(test)` is only activated when testing the current crate, not when building an upstream dependency.